src/go/k8s/cmd/configurator/main.go

```diff
@@ -323,7 +328,54 @@ func calculateRedpandaID(
 			return nil
 		}
 
-		redpandaID := int(rand.NewSource(time.Now().Unix()).Int63())
+		redpandaID := randID()
+
+		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+		defer cancel()
+
+		if len(c.hostName) < 3 {
+			return fmt.Errorf("hostname should have more than 3 caracters")
+		}
+		cmName := renderStsName(c.hostName, initialID)
+
+		cm := &corev1.ConfigMap{}
+		err = k8sClient.Get(ctx, client.ObjectKey{
+			Name: fmt.Sprintf("%s-ids", cmName),
+		}, cm)
+		if err != nil && k8sErrors.IsNotFound(err) {
+			cm.Name = fmt.Sprintf("%s-ids", cmName)
+			cm.Data = map[string]string{
+				redpandaIDs: "",
+			}
+			err = k8sClient.Create(ctx, cm)
```

Comment:
Have you considered reversing the roles here. I.e. managing the `-ids` configmap from the operator and reading it in the configurator.

The IDS would be created uniquely by the operator itself and also the configurator could use that cm to get info on other replicas. The nodes could still treat it as a "suggestion" and still report a different `nodeId` in the node_config...

Just thinking loud...

Reply:
Operator should track every PV that was bound to each POD and map each Redpanda ID to PV.

We don't have this information before statefuls set control schedules new POD. Then we can update configmap and configurator need to wait for this ID.